### PR TITLE
Add unit tests for ota.c

### PR DESCRIPTION
**Pull request by yanjos-dev**
* Add tests for the PAL failing while receiving a block.
* Test ingestDataBlockCleanup with a NULL file.
* Fix the downgrade test.
* Add a test for when the new version of the software is the same as what is currently running.
* Test receiving a job doc with a non-default file type.
* Remove unreachable code from ota.c.
* Test NULL inputs to verifyActiveJobStatus

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by gkwicker — APPROVED** [review]

File: test/unit-test/ota_utest.c [hunk 1]
```diff
@@ -2048,6 +2125,33 @@ void test_OTA_SelfTestJobDowngrade()
...
+
+    otaGoToState( OtaAgentStateWaitingForJob );
+    TEST_ASSERT_EQUAL( OtaAgentStateWaitingForJob, OTA_GetState() );
+
+    otaReceiveJobDocument();
+    receiveAndProcessOtaEvent();
+    TEST_ASSERT_EQUAL( true, resetCalled );
+}
+
+void test_OTA_SelfTestJobFileType()
```
Comment (divekarshubham) [inline test/unit-test/ota_utest.c:2140]: Might want to rename to reflect the condition tested?  `test_OTA_SelfTestJobNonFirmwareFileType`? or add a comment with the scenario.

File: test/unit-test/ota_utest.c [hunk 2]
```diff
@@ -1727,6 +1755,55 @@ void test_OTA_ReceiveLastFileBlockTooSmall()
     TEST_ASSERT_EQUAL( OtaAgentStateWaitingForJob, OTA_GetState() );
 }
 
+void test_OTA_ReceiveWithNullFile()
```
Comment (divekarshubham) [inline test/unit-test/ota_utest.c:1758]: could you add a comment with what happens when you receive a null file pointer?

**Review by divekarshubham — APPROVED** [review]

File: test/unit-test/ota_utest.c (hunk 1 above)
Comment (yanjos-dev) [reply test/unit-test/ota_utest.c]: done

File: test/unit-test/ota_utest.c (hunk 2 above)
Comment (yanjos-dev) [reply test/unit-test/ota_utest.c:1758]: done

**Review by gkwicker — APPROVED** [review]

**Pull request merged**